hnn_core/network.py

```diff
@@ -586,23 +612,28 @@ def _attach_drive(self, name, drive, weights_ampa, weights_nmda, location,
                         'synaptic_delays is either a common float or needs '
                         'to be specified as a dict for each cell type')
 
-        # this is needed to keep the drive GIDs identical to those in HNN,
-        # e.g., 'evdist1': range(272, 542), even when no L5_basket cells
-        # are targeted (event times lists are empty). The logic in
-        # _connect_celltypes is hard-coded to use these implict gid ranges
-        if self._legacy_mode:
```

Comment:
changes like this should be documented in `whats_new`. Are you also going to remove this parameter from `Network` instantiation? Is Stephanie on board with this? I remember we spent a lot of effort in trying to match our outputs with GUI outputs in the past.

Reply:
This one probably goes into the "Bug" section. Anything that impacts the users scripts/outputs should be documented in `whats_new`

Reply:
> changes like this should be documented in `whats_new`. 

Agreed. I'll update whats_new again once we all agree that this is the direction we want to move. I've reverted this PR to WIP for now.

> Are you also going to remove this parameter from `Network` instantiation? 

Not yet.  The `Network._legacy_mode` attribute is [still used in other places](https://github.com/jonescompneurolab/hnn-core/blob/40b07726a747c336debfe3a2683ee1607391a75d/hnn_core/network.py#L508-L515) that I wasn't planning on touching in this PR.

> Is Stephanie on board with this? I remember we spent a lot of effort in trying to match our outputs with GUI outputs in > the past.

My most recent commits do not get rid of legacy mode entirely, but do, however, change the convention for how many gids get instantiated. This will mess with our ability to reproduce our ground-truth `dpl.txt` simulation, but I was under the impression that we were okay with modifying ground-truth provided that we carefully control and understand exactly what is being changed.

Reply:
> I was under the impression that we were okay with modifying ground-truth provided that we carefully control and understand exactly what is being changed.

right, but we do lose the ability to reproduce the GUI results. Which is fine by me ... I guess my point is that, even if we carefully control things, we don't really want to go down the rabbit hole of figuring out what are the 3 commits that are different between the GUI and HNN-core. It's a very inefficient and time-consuming process of bug hunting

> Not yet. The Network._legacy_mode attribute is still used in other places that I wasn't planning on touching in this PR.

humm ... do you think we could leave out `_legacy_mode` from this PR and remove it entirely in a separate PR? It looks like those other places are just doing checks. Presumably because we don't want errors to be thrown in the GUI?

Reply:
Didn't we lose the ability to reproduce the GUI results in #314?

Reply:
well that's only a numerical issue thing. I think we can still get pretty much the same output

Reply:
Okay, I've restored legacy mode to its former glory in https://github.com/jonescompneurolab/hnn-core/pull/383/commits/3b3177828b4c0d3808388d89568ab2416114fa7d.

Reply:
Upon attempting to build the documentation, I just remembered why I removed legacy mode from `_attach_drive()` to begin with. @jasmainak do you have any time today to discuss this over zoom? I think our discuss will go faster that way.

Reply:
yeah I can talk in 15 minutes if you like